jconstraints-cvc4/src/test/java/io/github/tudoaqua/jconstraints/cvc4/solver/UNSATCoreTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2015 United States Government, as represented by the Administrator
+ *                of the National Aeronautics and Space Administration. All Rights Reserved.
+ *           2017-2021 The jConstraints Authors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.tudoaqua.jconstraints.cvc4.solver;
+
+import static gov.nasa.jpf.constraints.expressions.LogicalOperator.AND;
+import static gov.nasa.jpf.constraints.expressions.LogicalOperator.EQUIV;
+import static gov.nasa.jpf.constraints.expressions.LogicalOperator.IMPLY;
+import static gov.nasa.jpf.constraints.expressions.LogicalOperator.OR;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertTrue;
+
+import gov.nasa.jpf.constraints.api.ConstraintSolver;
+import gov.nasa.jpf.constraints.api.ConstraintSolver.Result;
+import gov.nasa.jpf.constraints.api.Expression;
+import gov.nasa.jpf.constraints.api.SolverContext;
+import gov.nasa.jpf.constraints.api.UNSATCoreSolver;
+import gov.nasa.jpf.constraints.api.Variable;
+import gov.nasa.jpf.constraints.expressions.Negation;
+import gov.nasa.jpf.constraints.expressions.PropositionalCompound;
+import gov.nasa.jpf.constraints.types.BuiltinTypes;
+import io.github.tudoaqua.jconstraints.cvc4.AbstractCVC4Test;
+import io.github.tudoaqua.jconstraints.cvc4.CVC4SMTCMDSolver;
+import io.github.tudoaqua.jconstraints.cvc4.CVC4Solver;
+import java.util.HashMap;
+import java.util.List;
+import org.testng.annotations.Test;
+
+public class UNSATCoreTest extends AbstractCVC4Test {
+
+  @Test
+  public void example1Test() {
+    UNSATCoreSolver cvc4UnsatCore = (UNSATCoreSolver) cvc4;
+    cvc4UnsatCore.enableUnsatTracking();
+    SolverContext ctx = cvc4.createContext();
```

Comment:
Upcasting requires downcasting in the next line
```suggestion
    CVC4SMTCMDSolver cvc4 = new CVC4SMTCMDSolver();
    cvc4.enableUnsatTracking();
```